### ISSUE TITLE
Graduate from standAlone to traditional cloud/server deployment

### ISSUE THREAD

**Issue by tangollama** [HospitalRun/hospitalrun-frontend#1048]
When someone downloads the electron app version of HospitalRun, they may want to "graduate" the single, desktop instance of the application to a traditional server-deployed model where the desktop instance is talking to an appliance or cloud server.

The application doesn't currently have a pathway to move from desktop instance to a traditional serve deployment. We need to that.

# Potential workflow
- Setup screen that allows user to login to AWS (if they don't have an account drives them outside of the app to setup an account
- If logged into AWS, allows a one-click deployment of the latest docker image of hospitalrun to be deployment to an EC2 instance
- Configures the Electron app to now consider that cloud instance the sync point for PouchDB to upload/sync data to the relater CouchDB instance.
- That configuration information is sharable to other Electron deployments

#Potential features
- [ ] Menu option in Gear on Electron deployment to upgrade to an cloud-backed deployment
- [ ] Menu-accessible setup screen that allows user to login to AWS (or links outside to account setup in AWS)
- [ ] From there, a one-click deployment process of an EC2 instance that houses the docker image (and related dependencies
- [ ] Stored configuration of the CouchDB instance that this local instance is syncing to
- [ ] Configuration screen for the CouchDB instance so that it can be setup

## Note
I used AWS b/c that's what I'm using these days. I'm cloud provider agnostic on this. The point is that people in developing settings need to be able to access the service.

**Comment by tangollama** [comment]
Once #1058 is merged into master, we could pursue this in earnest.

**Comment by jeremyklein** [comment]
I am interested in getting involved with HospitalRun, and this enhancement sounds very interesting to me. Has there been any progress towards this? Is there any documentation of the traditional server-deployed model that might help drive requirements?

**Comment by tangollama** [comment]
@jeremyklein thanks for reaching out and expressing an interest in working on this. There is a dockerized deployment of the app that can serve as one of the building blocks. After that, it might be most effective to talk it out amongst the two of us and then I'll document the plan better in this issue. Can you find me on the project Slack. I'm tangollama everywhere. :)

**Comment by fox1t** [comment]
@tehKapa here @tangollama suggests some nice workflows for the final users. I think that the same path is achievable on other cloudproviders/on-premise if something like openstack is used.
This task deserves some more exploration.
